invoices/invoiceregistry.go

```diff
@@ -880,11 +840,22 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 
 	switch invoiceHtlc.State {
 	case channeldb.HtlcStateCanceled:
-		return NewFailureResolution(
-			circuitKey, acceptHeight, result,
-		), nil
+		failureResolution := resolution.(*HtlcFailResolution)
```

Comment:
Shouldn't we just switch on the `resolution` itself now that it is complete? We previously inspected the invoice htlc mainly because we had not strong resolution outcome types, I think...